src/main/java/org/bitcoin/Secp256k1Context.java

```diff
@@ -17,42 +17,46 @@
 package org.bitcoin;
 
 /**
- * This class holds the context reference used in native methods 
+ * This class holds the context reference used in native methods
  * to handle ECDSA operations.
  */
 public class Secp256k1Context {
-  private static final boolean enabled; //true if the library is loaded
-  private static final long context; //ref to pointer to context obj
+    private static final boolean enabled; // true if the library is loaded
+    private static final long context; // ref to pointer to context obj
+    private static final Throwable loadError; // null, if library has been loaded successfully, otherwise - holds error details
 
-  static { //static initializer
-      boolean isEnabled = true;
-      long contextRef = -1;
-      try {
-          if ("The Android Project".equals(System.getProperty("java.vm.vendor"))) {
-              System.loadLibrary("co/rsk/secp256k1");
-          } else {
-              Secp256k1Loader.initialize();
-          }
-          contextRef = secp256k1_init_context();
-      } catch (UnsatisfiedLinkError e) {
-          System.out.println("Cannot load secp256k1 native library: " + e.toString());
-          isEnabled = false;
-      } catch (Exception e) {
-          System.out.println("Cannot load secp256k1 native library: " + e.toString());
-          isEnabled = false;
-      }
-      enabled = isEnabled;
-      context = contextRef;
-  }
+    static { //static initializer
+        boolean isEnabled = true;
+        long contextRef = -1;
+        Throwable error = null;
+        try {
+            if ("The Android Project".equals(System.getProperty("java.vm.vendor"))) {
```

Comment:
i guess this can be removed (in another PR)

Reply:
I don't really know why this check is needed. Needs some analysis. And yes, it's part of another issue